chaoscenter/subscriber/pkg/k8s/log.go

```diff
@@ -96,32 +86,84 @@ func (k8s *k8sSubscriber) CreatePodLog(podLog types.PodLogRequest) (types.PodLog
 
 // SendPodLogs generates graphql mutation to send events updates to graphql server
 func (k8s *k8sSubscriber) SendPodLogs(infraData map[string]string, podLog types.PodLogRequest) {
-	// generate graphql payload
-	payload, err := k8s.GenerateLogPayload(infraData["INFRA_ID"], infraData["ACCESS_KEY"], infraData["VERSION"], podLog)
+
+	logDetails, err := k8s.GetPodLogs(infraData, podLog)
 	if err != nil {
-		logrus.WithError(err).Print("Error while retrieving the workflow logs")
+		logrus.WithError(err).Print("failed to get pod logs")
 	}
+
+	// MarashalGQLData
+	processed, err := k8s.gqlSubscriberServer.MarshalGQLData(logDetails)
+	if err != nil {
+		processed = " Could not get logs "
+	}
+
+	// Generate graphql payload
+	payload, _ := k8s.GenerateLogPayload(infraData["INFRA_ID"], infraData["ACCESS_KEY"], infraData["VERSION"], processed, podLog)
+
 	body, err := k8s.gqlSubscriberServer.SendRequest(infraData["SERVER_ADDR"], payload)
 	if err != nil {
 		logrus.Print(err.Error())
 	}
 	logrus.Print("Response from the server: ", body)
 }
 
-func (k8s *k8sSubscriber) GenerateLogPayload(cid, accessKey, version string, podLog types.PodLogRequest) ([]byte, error) {
-	infraID := `{infraID: \"` + cid + `\", version: \"` + version + `\", accessKey: \"` + accessKey + `\"}`
-	processed := " Could not get logs "
+func (k8s *k8sSubscriber) GetPodLogs(infraData map[string]string, podLog types.PodLogRequest) (types.PodLog, error) {
+	logDetails := types.PodLog{}
+	mainLog, err := k8s.GetLogs(podLog.PodName, podLog.PodNamespace, "main")
+
+	// try getting argo pod logs
+	if err != nil {
+
+		// fetch ExperimentRun Phase from graphql
+		experimentRun, _ := k8s.gqlSubscriberServer.SendExperimentRunRuquest(infraData, podLog)
+		logDetails.MainPod = k8s.categorizeLogByPhase(experimentRun.Data.ExperimentRun.Phase)
+
+		logrus.WithError(err).Print("Failed to get pod logs for attempting to fetch experiment run phase")
+
+	} else {
+
+		logDetails.MainPod = strconv.Quote(strings.Replace(mainLog, `"`, `'`, -1))
+		logDetails.MainPod = logDetails.MainPod[1 : len(logDetails.MainPod)-1]
+
+		logDetails, err = k8s.CreatePodLog(infraData, logDetails, podLog)
 
-	// get the logs
-	logDetails, err := k8s.CreatePodLog(podLog)
-	if err == nil {
-		// process log data
-		processed, err = k8s.gqlSubscriberServer.MarshalGQLData(logDetails)
 		if err != nil {
-			processed = " Could not get logs "
+			return logDetails, err
 		}
 	}
 
+	return logDetails, nil
+}
+
+// Categorizing log messages by experiment phase
+func (k8s *k8sSubscriber) categorizeLogByPhase(phase string) string {
+	switch phase {
+	case "Completed":
+		return "Experiment pod is deleted"
```

Comment:
There can be a situation where users haven't added the cleanup chaos step, in that case, the experiment resources won't be deleted. We have to handle that situation as well

Reply:
@Saranya-jena I think this function will work in that case, because it will get data from mongoDB when getting the pod logs fails, but I will test it for that case.

Reply:
@Saranya-jena I remove the podGc field and the cleanup chaos step. Is this the correct procedure?

<img width="1190" alt="image" src="https://github.com/user-attachments/assets/94d47fe9-d49f-48b1-b8d8-fab9b8d9cab5">